src/lib.rs

```diff
@@ -166,9 +169,13 @@ pub trait DataNftMint:
 
         let treasury_address = self.treasury_address().get();
 
+        let bond_amount = self.get_bond_amount_for_lock_period(lock_period_sec);
+
         if price >= BigUint::zero() {
-            // @TODO ideally here we require that the payment is price + bondAmount (implement proxy view for bondAmount from the bonding contract)
-            // ... for now, the TX will fail if the payment.amount does not cover the price (anti_spam_tax) + bondAmount
+            require!(
+                payment.amount >= &price + &bond_amount,
```

Comment:
Why is it >= here, but = in the lower else?

Reply:
It should be ==, thanks for the catch